### PR TITLE
Improved haddocks of FetchDecline

### LINKED ISSUE #3927: Add haddocks for 'FetchDecline' data type

### PR DESCRIPTION
**Pull request by coot**
# Description

Fixes #3927 

# Checklist

- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Commits have useful messages
    - [ ] New tests are added if needed and existing tests are updated
    - [ ] If this branch changes Consensus and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../ouroboros-consensus/docs/interface-CHANGELOG.md)
    - [ ] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [ ] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [x] Self-reviewed the diff
    - [x] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [x] Reviewer requested

### CODE REVIEW

File: ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs [hunk 1]
```diff
@@ -112,16 +112,77 @@ type FetchDecision result = Either FetchDecline result
...
+     --
    | FetchDeclineChainNoIntersection
+
+     -- | The chain has already been fetched.
+     --
    | FetchDeclineAlreadyFetched
+
+     -- | The chain is already being downloaded.
+     --
    | FetchDeclineInFlightThisPeer
```
Comment (bolt12) [inline ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:131]: What's the difference between ThisPeer vs OtherPeer?

**Review by bolt12 — APPROVED** [review]

File: ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs (hunk 1 above)
Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:131]: we have two filtering functions:
* `filterNotAlreadyInFlightWithOtherPeers` - in deadline mode it allows to download the same block concurrently from different peers, in block fetch mode it filters out blocks that we are already fetching from other peers
* `filterNotAlreadyInFlightWithPeer` - protects us from starting fetching a block from a peer from which we are already downloading it.

Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:131]: I slightly improved the haddocks.

Comment (bolt12) [reply ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs:131]: Thank you!

**Comment by coot** [comment]
bors merge

**Pull request merged**